src/Api/Service.php

```diff
@@ -517,6 +517,7 @@ public function getDefinition()
     public function setDefinition($definition)
     {
         $this->definition = $definition;
+        $this->shapeMap = new ShapeMap($definition['shapes']);
```

Comment:
Correct- the original `shapeMap` is constructed with the original definition, so when the definition is updated, the shapeMap should also be updated if there have been modifications made to any shapes in the definition.  Because we don't currently have a use case for updating the `shapeMap` independent of model definition changes (and customers don't either), I don't think we should add a separate method. This can always be added later if it turns out we need it.

Reply:
Understood. Is just that the name of the method does not refer to the only behavior contained within. But, I guess that since is a MUST that shapeMap gets updated when definition it is also updated then, I think it should be ok.